extensions/omni_txn/docs/retry.md

```diff
@@ -0,0 +1,50 @@
+# Transaction Retry
+
+When using serializable transactions, it's often necessary to employ a retry strategy in case of serialization failure.
+The algorthms
+are fairly typical, so repeating them manually doesn't always make a lot of sense. `omni_txn` provides `retry` procedure
+to handle such typical cases.
+
+|        Parameter | Type | Description                                                          |
+|-----------------:|------|----------------------------------------------------------------------|
+|        **stmts** | text | Statement(s) to execute. Multiple statements separated by semicolon. |
+| **max_attempts** | int  | Max number of times to retry. 0 means no retries. 10 by default.     |
+
+## Retry attempt
+
+There is a helper function `omni_txn.current_retry_attempt()` that provides retry attempt during the `retry()` call. 0
+stands
+for the first run, 1 for the first retry, etc.
+
+## Example
+
+Let's consider the following schema:
+
+```postgresql
+create table inventory
+(
+    id           serial primary key,
+    product_name text,
+    quantity     int
+);
+insert into inventory (product_name, quantity)
+values ('Widget', 100);
+```
+
+Now, if we have these two simultaneous transactions happening, the second one may have committed first:
+
+```postgresql
+--- (1)
+call omni_txn.retry($$
+select quantity from inventory where product_name = 'Widget';
+update inventory set quantity = quantity + 20
+       where product_name = 'Widget'
+$$);
```

Comment:
Seems useful! Have you thought of a global configuration? When enabling serializable isolation is usually cumbersome to retry individual statements, it'd be more convenient to have a GUC.. with that it can be enabled per session, role or db.